File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.js
const fs = require("fs");

const stdin = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const compare = (a, b) => {
  const [age_a, age_b] = [parseInt(a[0]), parseInt(b[0])];

  if (age_a === age_b) {
    return a[2] - b[2];
  }

  return age_a - age_b;
};

const data = stdin
  .slice(1)
  .map((row) => row.split(" "))
  .map((line, i) => {
    const [age, name] = line;
    return [parseInt(age), name, i];
  })
  .sort(compare);

console.log(data.map((row) => row.splice(0, 2).join(" ")).join("\n"));

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.cc
#include <bits/stdc++.h>

using namespace std;

struct member {
  int order;
  int age;
  string name;
};

bool compare(member a, member b) {
  if (a.age == b.age) return a.order < b.order;
  return a.age < b.age;
}

int main() {
  ios_base::sync_with_stdio(0);
  cin.tie(0);

  int n;
  cin >> n;

  member members[n];

  for (int i = 0; i < n; i++) {
    member m;
    cin >> m.age >> m.name;
    m.order = i;
    members[i] = m;
  }

  sort(members, members + n, compare);

  for (int i = 0; i < n; i++) {
    cout << members[i].age << ' ' << members[i].name << '\n';
  }

  return 0;
}
